### PR TITLE
Updating ER to use new reblog behavior

### LINKED ISSUE #407: Return to the olds chains... or make them scrollable
Follow official Tumblr staf page, the next regression is this:
http://support.tumblr.com/post/128127169452/ever-see-a-crazy-long-indecipherable-reblog-chain
This will make chains literally exponentially longers like those from RP blogs.

I post this here before to let know this and prevent complains when the update break the add-ons related to chains.

### PR DESCRIPTION
**Pull request by blackjackkent**
Fixes #407

### CODE REVIEW

File: Extensions/editable_reblogs.js [hunk 1]
```diff
@@ -29,23 +30,22 @@ XKit.extensions.editable_reblogs = new Object({
...
-		if (top_quote_link.length > 0) {
-			$(top_quote_link[0]).addClass('tumblr_blog');
-		}
-		quote_links.each(function() {
-			$(this).addClass('tumblr_blog');
+		reblog_tree.find( ".reblog-list-item" ).each(function( index ) {
+			var reblog_data = {
+				reblog_content: $(this).find('.reblog-content').html(),
+				reblog_author: $(this).find('.reblog-tumblelog-name').text(),
+				reblog_url: $(this).find('.reblog-tumblelog-name').attr('href')
```
Comment (Wolvan) [inline Extensions/editable_reblogs.js]: Semicolon missing here

File: Extensions/editable_reblogs.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 //* TITLE Editable Reblogs **//
-//* VERSION 1.0.9 **//
+//* VERSION 2.0.0 **//
 //* DESCRIPTION	Restores ability to edit previous reblogs of a post **//
-//* DEVELOPER dlmarquis **//
+//* DEVELOPER dlmarquis, blackjackkent **//
```
Comment (Wolvan) [inline Extensions/editable_reblogs.js]: If you change the Developer, change it to new-xkit

File: Extensions/editable_reblogs.js (hunk 1 above)
Comment (bvtsang) [inline Extensions/editable_reblogs.js]: I don't think a value within an object needs a semicolon at the end.

**Comment by nightpool** [comment]
407 was closed fyi (didn't have a clear resolution given we hadn't even seen the update at the time.

File: Extensions/editable_reblogs.js (hunk 1 above)
Comment (blackjackkent) [inline Extensions/editable_reblogs.js]: It's the line below it -- where the object ends.

Fixed.

Comment (Wolvan) [inline Extensions/editable_reblogs.js]: Yeah, clicked the wrong line, it was meant for line 38

**Comment by Wolvan** [comment]
Can't test right now, but it LGTM so far

**Comment by bvtsang** [comment]
I don't know if this matters, but for people who don't have the update yet, when they click reblog the input text area will be blank.

**Comment by Wolvan** [comment]
I'd say that's a temporary problem that resolves itself once the update rolled out completely, @bvtsang

**Comment by nightpool** [comment]
Can we add a check anyway? If we're looking at pushing this out tonight then we should make sure to account for the fact that not everybody may be migrated

File: Extensions/editable_reblogs.js [hunk 3]
```diff
@@ -29,23 +30,22 @@ XKit.extensions.editable_reblogs = new Object({
...
+			var reblog_data = {
+				reblog_content: $(this).find('.reblog-content').html(),
+				reblog_author: $(this).find('.reblog-tumblelog-name').text(),
+				reblog_url: $(this).find('.reblog-tumblelog-name').attr('href')
+			};
+			all_quotes.push(reblog_data);
+		});
+		var all_quotes_text = "";
+		all_quotes.forEach(function(data, index, all) {
+			all_quotes_text = "<a href='" + data.reblog_url + "'>" + data.reblog_author + "</a><blockquote>" + all_quotes_text + data.reblog_content + "</blockquote>";
```
Comment (nightpool) [inline Extensions/editable_reblogs.js]: should have class="tumblr_blog"

**Comment by Wolvan** [comment]
Reblogging a ER reblog breaks the caption
image

**Comment by Wolvan** [comment]
The way I encountered the problem is as follows:
1.) Reblog a post that has the new design from my dash (I reblogged to wolvan-test)
2.) Click the reblog button on my reblog
3.) Don't change anything in the editor
4.) Click reblog
5.) See the caption deleted, except of the wolvan-test link from before

**Comment by 0xazure** [comment]
Possibly partially related to the issue Wolvan demonstrated above, but editing your own reblog also makes previous post text disappear.

Repro steps:
- Reblog a post and add some text to it
- Refresh dash
- Edit the newly reblogged post, but _edit_ the text you added when you first reblogged it, don't add anything on a new line
- Save & refresh dash
- Notice there's text missing from the person you reblogged from.

The visual aspect of it also doesn't seem to work on _all_ posts.  It handles most of them fine, but specific examples like http://new-xkit-discussion.tumblr.com/post/128212261064/katisconfused-familiaralien-virvendir don't display with the old blockquotes-style as they should.

Tested on OS X, latest Firefox.

**Comment by 0xazure** [comment]
Not sure if this is a limitation in how we're editing post bodies, but when a user reblogs a post without adding anything, it's treated like they added a null string.  That is, it assumes that a user who's reblogging a post is going to add text to the post, and automatically nests the post in the post editor _before_ completing the reblog.

To reproduce:
- Make a new text post (haven't tested with other post types yet) with some kind of body (with or without title)
- Refresh dash
- Click reblog to reblog from yourself without adding any text
- Before clicking 'reblog', notice the extra level of nesting in the rich text editor.

**Comment by blackjackkent** [comment]
Possible new approach (Current post window display on the left, proposed ER display on the right...)

hr-test

**Comment by 0xazure** [comment]
Commenting for posterity: issues resolved, lgtm on OS X, latest Firefox.

**Comment by Wolvan** [comment]
I agree with @0xazure: issues resolved, lgtm, latest Chrome on Windows
@homu r+

**Pull request merged**